### ISSUE TITLE
Lode launcher wont launch mac

### ISSUE THREAD

**Issue by Kevmate** [lightning-power-users/node-launcher#208]
* **I'm submitting a ...**
  - [ ] bug report
  - [ ] feature request
  - [ ] support request

* **What is the current behavior?**

* **If the current behavior is a bug, please provide the steps to reproduce**

* **What is the expected behavior?**

* **What is the motivation / use case for changing the behavior?**

* **Please tell us about your environment:**

  - Operating System: [ Windows | macOS | Linux ]

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

**Comment by Kevmate** [comment]
I find that when I start node launcher it just doesn't open. No error messages, just fails. I installed it a few weeks ago and it worked then, but for the last couple of weeks it doesn't launch. I have tried deleting and re-installing, no joy. I am running a bitcoin full node on mainnet, that always works. OS is High Sierra. I have no other problems with the machine that I know of.

**Comment by Kevmate** [comment]
I had a search, and watched the system error logs. Here is the error that it looks like I am getting from the Node Launcher when I run it:

Mar  2 16:54:36 Kevins-MacBook com.apple.xpc.launchd[1] (com.apple.imfoundation.IMRemoteURLConnectionAgent): Unknown key for integer: _DirtyJetsamMemoryLimit
Mar  2 16:54:37 Kevins-MacBook com.apple.xpc.launchd[1] (Node Launcher.10784[795]): Service exited with abnormal code: 255
Mar  2 16:54:37 Kevins-MacBook systemstats[50]: assertion failed: 17G5019: systemstats + 914800 [D1E75C38-62CE-3D77-9ED3-5F6D38EF0676]: 0x40
Mar  2 16:54:43 Kevins-MacBook kcm[800]: DEPRECATED USE in libdispatch client: Setting timer interval to 0 requests a 1ns timer, did you mean FOREVER (a one-shot timer)?

I'm not sure if these are all yours, but the last one certainly comes up every time.

**Comment by PierreRochard** [comment]
Thanks for reporting it! This is a weird one, I'll see if others have the same issue

**Comment by tigerhbkim** [comment]
I'm having the same issue here

**Comment by nk1tz** [comment]
I came to report this same issue.

**Comment by SvenStarfury** [comment]
yes, i posted a similar issue: https://github.com/lightning-power-users/node-launcher/issues/204
plz halp! =) ∞ <3
(and thank you for your service, good sir!)

**Comment by Kevmate** [comment]
#204 looks a bit different. Not the same crash, from what I can see. Also I am on macbook, not mac pro. Not sure if that matters. I would really like to know what to try here. There must be smething different about my install.

**Comment by mguttentag** [comment]
Hi, I posted how I solved what sounds like a similar issue https://github.com/lightning-power-users/node-launcher/issues/213#issuecomment-476923666 -- may consider trying this? (comment reprinted below): 

I was having similar issues (Node Launcher bouncing in dock then disappearing + not starting) and appear to have solved it! I should note that previously I was also having an issue getting Bitcoin Core to open on its own, it would display, "Error: Cannot parse configuration file: specified data directory..." -- but it was opening fine through Node Launcher, at least until I restarted my Mac and then Node Launcher wouldn't boot. Anyway, it's working now and here's what I did in case it's helpful:

**tl;dr: I moved my Bitcoin datadir somewhere else (outside of "Library") then modified the .conf file and bitcoin settings to point to the new location, then everything worked again.**

   

1.  Moved my "bitcoin" data directory out of the Library (I moved it from Users/**/Library/Application\ Support/Bitcoin over to Users/**/Bitcoin)
2. Edited Bitcoin.conf so data directory read the new location of Bitcoin folder (Users/**/Bitcoin)
    [here I also used Time Machine to restore an earlier version of Bitcoin-Qt, i.e. from a few days ago, but I do not think that was necessary. Same with Node Launcher, but they were otherwise the same versions of the program]
3. Opened Terminal.app and ran /Applications/Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt --resetguisettings
4. When the "Welcome to Bitcoin Core" dialogue came up, I clicked "Use a custom data directory" and directed it to the new location of the Bitcoin folder (Users/**/Bitcoin).

5. Then Bitcoin Core seemed to be working, so I quit it. I then launched Node Launcher. Node Launcher now loaded. When I pressed "Launch Bitcoin" in Node Launcher, it launched, and LND worked too.

Let me know if that works for your problem too!

**Comment by tkijewski** [comment]
Same issue

```
Apr 13 14:55:55 MacBook-Pro com.apple.xpc.launchd[1] (com.apple.imfoundation.IMRemoteURLConnectionAgent): Unknown key for integer: _DirtyJetsamMemoryLimit
Apr 13 14:55:57 MacBook-Pro com.apple.xpc.launchd[1] (Node Launcher.5304[16407]): Service exited with abnormal code: 255
```

**Comment by PierreRochard** [comment]
Unfortunately my 10.12 macOS machine crapped out and isn't booting, so I've been building the binary with 10.14 and I think that's causing this problem. I'm working on replacing the 10.12 machine (10.10 would be nice), until then you'll have to follow the developer instructions and build it yourself :(

**Comment by tkijewski** [comment]
Can confirm building from source does work
